### ISSUE TITLE
Oracle data chart

### ISSUE THREAD

**Issue by thegreat-vanderlinde** [vyper-protocol/vyper-otc-ui#313]
We have to include in the contracts summary page a chart reporting historical oracles data.

We have to research some APIs that can grant this kind of information, we can evaluate both free and paid ones.

- [ ] research oracles APIs for historical data. We can evaluate solutions for data only where we'll build the UI around or simpler ones where we can use some GUI ready-to-use components
- [ ] if needed we can add API references to the JSON dataset, blocked by #311
- [ ] add the char in the summary page, UI look and feel is still an open discussion. see below comments

**Comment by sudoshreyansh** [comment]
I researched more into this. We can use the TradingView chart widget for the chart. For the chart's data, I looked into 3 options:
- Off-chain indexing: This is how Mango Markets work. They have [Serum History](https://github.com/blockworks-foundation/serum-history) which creates data store in redis database and then use this to display the chart. But we can then store the data only for prior known oracles and not any external oracle.
- Transaction history: We can use the transaction history of the oracle programs to get a historical data feed for the chart.
- TradingView's data: TradingView also has crypto pricing data but the pricing data would be for a cryptocurrency rather than the oracle. 

@thegreat-vanderlinde The transaction history for a oracle seems to be most viable option for me. Shall I go with it?
